src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -132,21 +132,43 @@ public static ValueTask<SslStream> EstablishSslConnectionAsync(SslClientAuthenti
             }
 
             // Create the SslStream, authenticate, and return it.
-            return EstablishSslConnectionAsyncCore(new SslStream(stream), sslOptions, cancellationToken);
+            return EstablishSslConnectionAsyncCore(stream, sslOptions, cancellationToken);
         }
 
-        private static async ValueTask<SslStream> EstablishSslConnectionAsyncCore(SslStream sslStream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
+        private static async ValueTask<SslStream> EstablishSslConnectionAsyncCore(Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
         {
+            SslStream sslStream = new SslStream(stream);
```

Comment:
I'd purposefully done this at the call site rather than here. The state machine for this async method will now contain fields for both streams, rather than just one. But it dorsn't really matter